lib/block_view_access_groups.go

```diff
@@ -0,0 +1,964 @@
+	}
+	return nil
+}
+
+// ValidateKeyAndNameWithUtxo validates public key and key name, which are used in DeSo V3 Messages protocol.
+// The function first checks that the key and name are valid and then fetches an entry from UtxoView or DB
+// to check if the key has been previously saved. This is particularly useful for connecting V3 messages.
+func (bav *UtxoView) ValidateKeyAndNameWithUtxo(ownerPublicKey, messagingPublicKey, keyName []byte, blockHeight uint32) error {
+	// First validate the public key and name with ValidateGroupPublicKeyAndName
+	err := ValidateGroupPublicKeyAndName(messagingPublicKey, keyName)
+	if err != nil {
+		return errors.Wrapf(err, "ValidateKeyAndNameWithUtxo: Failed validating "+
+			"messagingPublicKey and keyName")
+	}
+
+	// Fetch the messaging key entry from UtxoView.
+	messagingGroupKey := NewAccessGroupKey(NewPublicKey(ownerPublicKey), keyName)
+	// To validate a messaging group key, we try to fetch the simplified group entry from the membership index.
+	messagingGroupEntry := bav.GetMessagingGroupForMessagingGroupKeyExistence(
+		messagingGroupKey, blockHeight, false)
+	if messagingGroupEntry == nil || messagingGroupEntry.isDeleted {
+		return fmt.Errorf("ValidateKeyAndNameWithUtxo: non-existent messaging key entry "+
+			"for ownerPublicKey: %s", PkToString(ownerPublicKey, bav.Params))
+	}
+
+	// Compare the UtxoEntry with the provided key for more validation.
+	if !reflect.DeepEqual(messagingGroupEntry.AccessPublicKey[:], messagingPublicKey) {
+		return fmt.Errorf("ValidateKeyAndNameWithUtxo: keys don't match for "+
+			"ownerPublicKey: %s", PkToString(ownerPublicKey, bav.Params))
+	}
+
+	if !EqualGroupKeyName(messagingGroupEntry.AccessGroupKeyName, NewGroupKeyName(keyName)) {
+		return fmt.Errorf("ValidateKeyAndNameWithUtxo: key name don't match for "+
+			"ownerPublicKey: %s", PkToString(ownerPublicKey, bav.Params))
+	}
+	return nil
+}
+
+// isMemberMuted returns true if the member is muted in the group.
+func (bav *UtxoView) isMemberMuted(
+	messagingGroupEntry *AccessGroupEntry, memberPublicKey *PublicKey) (bool, error) {
+	// Create enumeration key.
+	enumerationKey := NewGroupEnumerationKey(messagingGroupEntry.GroupOwnerPublicKey, messagingGroupEntry.AccessGroupKeyName[:], memberPublicKey)
+	// Check if enumerationKey exists in GroupMemberAttributes mapping.
+	muted, exists := bav.GroupMemberAttributes[*enumerationKey]
+	// If utxoView mapping exists, return value.
+	if exists {
+		return muted, nil
+	}
+
+	// If utxoView mapping doesn't exist, check DB.
+	isMuted, err := DBIsMemberMutedInGroupMembersIndex(bav.Handle, bav.Snapshot, messagingGroupEntry.GroupOwnerPublicKey, messagingGroupEntry.AccessGroupKeyName, memberPublicKey)
+	if err != nil {
+		return false, errors.Wrapf(err, "isMemberMuted: Problem checking if member is muted in DB")
+	}
+	return isMuted, nil
+}
+
+// setMemberMuted sets the muted status of a member in the group.
+func (bav *UtxoView) setMutedMembersMapping(
```

Comment:
can we refactor these functions to use `<groupOwnerPublicKey, groupKeyName>` as parameters instead of the `AccessGroupEntry`?

Reply:
done